queries/tact/highlights.scm

```diff
@@ -0,0 +1,311 @@
+; NOTE: Neovim's highlight order differs from Tree-sitter's and Helix's,
+; NOTE: as broad captures should be placed first, narrow queries - second.
+; See: https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md#highlights
+; ----------------------------------------------------------------------------------------------
+; variable
+; --------
+(identifier) @variable
+
+; variable.builtin
+; ----------------
+(self) @variable.builtin
+
+; variable.parameter
+; ------------------
+(parameter
+  name: (identifier) @variable.parameter)
+
+; punctuation.delimiter
+; ---------------------
+[
+  ";"
+  ","
+  "."
+  ":"
+  "?"
+] @punctuation.delimiter
+
+; punctuation.bracket
+; -------------------
+[
+  "("
+  ")"
+  "{"
+  "}"
+] @punctuation.bracket
+
+; operator
+; --------
+[
+  "-"
+  "-="
+  "+"
+  "+="
+  "*"
+  "*="
+  "/"
+  "/="
+  "%"
+  "%="
+  "="
+  "=="
+  "!"
+  "!="
+  "!!"
+  "<"
+  "<="
+  "<<"
+  ">"
+  ">="
+  ">>"
+  "&"
+  "|"
+  "^"
+  "&&"
+  "||"
+] @operator
+
+; constructor
+; -----------
+(instance_expression
+  name: (identifier) @constructor)
+
+(initOf
+  name: (identifier) @constructor)
+
+; type
+; ----
+(type_identifier) @type
+
+; type.builtin
+; ------------
+((identifier) @type.builtin
+  (#eq? @type.builtin "SendParameters"))
+
+(bounced_type
+  "bounced" @type.builtin
+  "<" @punctuation.bracket
+  ">" @punctuation.bracket)
+
+(map_type
+  "map" @type.builtin
+  "<" @punctuation.bracket
+  ">" @punctuation.bracket)
+
+((type_identifier) @type.builtin
+  (#any-of? @type.builtin "Address" "Bool" "Builder" "Cell" "Int" "Slice" "String" "StringBuilder"))
+
+(tlb_serialization
+  "as" @keyword.type
+  type: (identifier) @type.builtin
+  (#any-of? @type.builtin
+    "int8" "int16" "int32" "int64" "int128" "int256" "int257" "uint8" "uint16" "uint32" "uint64"
+    "uint128" "uint256" "coins" "remaining" "bytes32" "bytes64"))
+
+; string
+; ------
+(string) @string
+
+; string.escape
+; -------------
+(escape_sequence) @string.escape
+
+; string.special.path
+; -------------------
+(import_statement
+  library: (string) @string.special.path)
+
+; boolean
+; -------
+(boolean) @boolean
+
+; constant
+; --------
+(constant
+  name: (identifier) @constant)
+
+; constant.builtin
+; ----------------
+(null) @constant.builtin
+
+((identifier) @constant.builtin
+  (#any-of? @constant.builtin
+    "SendBounceIfActionFail" "SendPayGasSeparately" "SendIgnoreErrors" "SendDestroyIfZero"
+    "SendRemainingValue" "SendRemainingBalance" "ReserveExact" "ReserveAllExcept" "ReserveAtMost"
+    "ReserveAddOriginalBalance" "ReserveInvertSign" "ReserveBounceIfActionFail"))
+
+; property
+; --------
+(instance_argument
+  name: (identifier) @property)
+
+(lvalue
+  (_)
+  (_) @property)
+
+(field_access_expression
+  name: (identifier) @property)
+
+(trait_body
+  (constant
+    name: (identifier) @property))
+
+(contract_body
+  (constant
+    name: (identifier) @property))
+
```

Comment:
What do you think about:
```scheme
(contract
  name: (identifier) @type)
```
and
```scheme
(trait
  name: (identifier) @type)
```

Reply:
Also:
```query
(trait_list
  (identifier) @type)
```

Reply:
Hmm, I don't think that's right given that those are not really types. Moreover, I'm not sure if their highlighting for `initOf` will preserve those updates (kinda like `new` keyword in, say, Java, but instead of creating the instance of a thing it returns some specific data for it). I'd keep their name identifiers as `@variable` captures.